contracts/GovHub.sol

```diff
@@ -208,29 +211,34 @@ contract GovHub is System, IParamSubscriber {
 
   /// Execute the proposal
   /// @param proposalId The proposal Id
-  function execute(uint256 proposalId) public payable onlyInit {
+  function execute(uint256 proposalId) public payable nonReentrant onlyInit { 
     require(getState(proposalId) == ProposalState.Succeeded, "proposal can only be executed if it is succeeded");
     Proposal storage proposal = proposals[proposalId];
     proposal.executed = true;
     uint256 targetSize = proposal.targets.length;
     for (uint256 i = 0; i < targetSize; i++) {
       bytes memory callData;
       if (bytes(proposal.signatures[i]).length == 0) {
-        callData = proposal.calldatas[i];
+        // eth transfer
+        callData = proposal.calldatas[i]; 
       } else {
-        callData = abi.encodePacked(bytes4(keccak256(bytes(proposal.signatures[i]))), proposal.calldatas[i]);
+        // function call
+        bytes4 functionSelector = bytes4(keccak256(bytes(proposal.signatures[i])));
+        callData = abi.encodePacked(functionSelector, proposal.calldatas[i]);
       }
 
-      (bool success, bytes memory returnData) = proposal.targets[i].call{ value: proposal.values[i] }(callData);
+      //@openissue: is it guaranteed that all proposal targets will be platform contracts?
```

Comment:
I think it's reasonable to assume that all proposal targets will be platform contracts. I can't think of any cases where we'd want governance to control a parameter outside of the set of platform contracts. Can keep as @openissue for now.

Reply:
If that is the case we must assert it as part of a proposal acceptance!